condacolab.py

```diff
@@ -89,11 +89,38 @@ def install_from_url(
         stderr=STDOUT,
         text=True,
     )
+
+    # installing google.colab packages, matplortlib-base and psutil.
+
+    conda_exe = "mamba" if os.path.isfile(f"{prefix}/bin/mamba") else "conda"
+
+    psutil_task = run(
+        [f"{prefix}/bin/{conda_exe}", "install", "-yq", "matplotlib-base", "psutil", "google-colab"],
+        check=False,
+        stdout=PIPE,
+        stderr=STDOUT,
+        text=True,
+    )
+
+    colab_task = run(
+        [f"{prefix}/bin/python", "-m", "pip", "-q", "install", "https://github.com/googlecolab/colabtools/archive/refs/heads/main.zip", "condacolab"],
+        check=False,
+        stdout=PIPE,
+        stderr=STDOUT,
+        text=True,
+    )
+
     os.unlink(installer_fn)
     with open("condacolab_install.log", "w") as f:
         f.write(task.stdout)
+    with open("colab_task.log", "w") as f:
+        f.write(colab_task.stdout)
+    with open("psutil_task.log", "w") as f:
+        f.write(psutil_task.stdout)
     assert (
-        task.returncode == 0
+        task.returncode == 0 and
+        colab_task.returncode  == 0 and
+        psutil_task.returncode == 0
```

Comment:
```suggestion
        task.returncode == conda_task.returncode == pip_task.returncode == 0
```

Isn't Python cool here and there? 😎

Reply:
That said, we should probably split the checks one by one so we can tell the user which one failed.

I'd also do the checks one by one, after their corresponding subprocess has been run, to minimize compounding effects in the errors (something failed, which led the 2nd process to also fail, etc).